extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -625,10 +625,13 @@ export class AmpConsent extends AMP.BaseElement {
         const ampdoc = this.getAmpDoc();
         const sourceBase = getSourceUrl(ampdoc.getUrl());
         const resolvedHref = resolveRelativeUrl(href, sourceBase);
+        const xhrService = Services.xhrFor(this.win);
         return ampdoc.whenFirstVisible().then(() => {
-          return Services.xhrFor(this.win)
+          return xhrService
             .fetchJson(resolvedHref, init)
-            .then(res => res.json());
+            .then(res =>
+              xhrService.xssiJson(res, this.consentConfig_['stripPrefix'])
```

Comment:
have you considered a simpler API:

```js
fetchJson(href, init, xssiPrefix = undefined)
```

Reply:
As counterintuitive as it is, `fetchJson` does not actually return json :). It returns a `Promise<Response>`, which makes stripping the prefix pretty difficult.

Initially I had written a pretty cool hack where I did just as you suggest, and then created a cloned `Response` where the `.body` had the prefix stripped from it. The issue came with cloning all of Response's other properties, some of which are not writable.

Reply:
I see your pain :-).

What about introducing a new API in xhr-impl.js:

```
/**
  * @return {!Promise<?JsonObject>}
  */
fetchJsonObject(href, init, xssiPrefix = undefined)
```

we can gradually migrate existing calls to fetchJson over to this new method.

Reply:
Beginning an initiative to migrate from a Response returning `fetchJson` to a `JsonObject`returning one would be sizable. It has 36 references in src and another 199 in test code.

Its definitely a good idea, since I checked and it looks like 95%+ of its usages contain `.then(res => res.json())` as the next line.

Would you be okay with keeping the API as-is for this PR, and I'll explore potential migration paths in a followup PR?  I could imagine a first PR that creates the new function and renames all the current `fetchJson`'s to `fetchJsonLegacy`.

Reply:
That works.

Reply:
Perfect :)